src/ConnectedMode/UI/ProjectSelection/ProjectSelectionViewModel.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.ObjectModel;
+using SonarLint.VisualStudio.Core.WPF;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI.ProjectSelection;
+
+public class ProjectSelectionViewModel : ViewModelBase
+{
+    public ObservableCollection<ServerProject> ProjectResults { get; } = [];
+
+    public string ProjectSearchTerm
+    {
+        get => projectSearchTerm;
+        set
+        {
+            projectSearchTerm = value;
+            SearchForProject();
+            RaisePropertyChanged();
+        }
+    }
+
+    public ServerProject SelectedProject
+    {
+        get => selectedProject;
+        set
+        {
+            selectedProject = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsProjectSelected));
+        }
+    }
+
+    public bool IsProjectSelected => SelectedProject != null;
+
+    private string projectSearchTerm;
+    private ServerProject selectedProject;
+
+    public void InitProjects(List<ServerProject> projects)
+    {
+        ProjectResults.Clear();
+        projects.ForEach(x => ProjectResults.Add(x));
+    }
+
+    private void SearchForProject()
+    {
+        if (string.IsNullOrEmpty(ProjectSearchTerm))
```

Comment:
OnInitialize the ViewModel will be responsible for fetching the projects using SLCore. It will be simpler, in this case, to re-fetch the projects when the filter is cleared. I didn't implement that in this PR because it will require the SLCore logic or else I have to add some caching mechanism (a List probably) that will be removed on the next sprint.

Reply:
Created a task to not forget this point: https://sonarsource.atlassian.net/browse/SLVS-1350